source/features/deep-blame.tsx

```diff
@@ -0,0 +1,127 @@
+import './deep-blame.css';
+import blurAccessibly from 'blur-accessibly';
+import delegate, {DelegateEvent} from 'delegate-it';
+import mem from 'mem';
+import React from 'dom-chef';
+import versionIcon from 'octicon/versions.svg';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoGQL, getReference, looseParseInt} from '../libs/utils';
+
+const getPullRequestBlameCommit = mem(async (commit: string, prNumber: number): Promise<string | false> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			object(expression: "${commit}") {
+				... on Commit {
+					associatedPullRequests(last: 1) {
+						nodes {
+							number
+							mergeCommit {
+								oid
+							}
+							commits(last: 1) {
+								nodes {
+									commit {
+										oid
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const associatedPR = repository.object.associatedPullRequests.nodes[0];
+	const mergeCommit = associatedPR.mergeCommit.oid;
+
+	if (associatedPR.number === prNumber && mergeCommit === commit) {
+		return associatedPR.commits.nodes[0].commit.oid;
+	}
+
+	return false;
+});
+
+async function redirectToBlameCommit(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void> {
+	const blameLink = event.delegateTarget;
+	const blameHunk = blameLink.closest('.blame-hunk')!;
+	const lineNumber = select('.js-line-number', blameHunk)!.textContent!;
+	const prNumber = looseParseInt(select('.issue-link', blameHunk)!.textContent!);
+	const prCommit = select<HTMLAnchorElement>('a.message', blameHunk)!.href.split('/').pop()!;
+	if (blameLink.href && event.altKey) {
+		event.preventDefault();
+	}
+
+	const githubSpinner = (
+		<img
+			src="https://github.githubassets.com/images/spinners/octocat-spinner-128.gif"
+			className="mr-2"
+			width="18"
+		/>
+	);
+
+	blameLink.firstElementChild!.replaceWith(githubSpinner);
+	// Hide tooltip after click, it’s shown on :focus
+	blurAccessibly(blameLink);
+
+	const prBlameCommit = await getPullRequestBlameCommit(prCommit, Number(prNumber));
+	if (prBlameCommit) {
+		const href = new URL(location.href.replace(getReference()!, prBlameCommit));
+		href.hash = 'L' + lineNumber;
+		location.href = String(href);
```

Comment:
We have a link, let's use it to our advantage
```suggestion
		blameLink.pathname = location.pathname.replace(getReference()!, prBlameCommit);
		blameLink.hash = 'L' + lineNumber;
		blameLink.click();
```